proxy/src/main/java/com/velocitypowered/proxy/connection/client/ConnectedPlayer.java

```
@@ -1026,6 +1030,31 @@ void setClientBrand(final @Nullable String clientBrand) {
     this.clientBrand = clientBrand;
   }
 
+  @Override
+  public void playSound(@NotNull Sound sound, @NotNull Sound.Emitter emitter) {
+    Preconditions.checkNotNull(sound, "sound");
+    Preconditions.checkNotNull(emitter, "emitter");
+    Preconditions.checkArgument(emitter.equals(Sound.Emitter.self()), "non-self emitter not supported");
```

Better to fail silently, that's more adventure

---

Or possibly just ignore the emitter and play normally

---

Also, any reason you only override play sound with an emitter?

---

> Also, any reason you only override play sound with an emitter?

Yes, because the other one indicates it uses the position of the player, which wouldn't follow it. So I moved the implemented method [here](https://github.com/PaperMC/Velocity/pull/1422/commits/6ec69dc8b3a6223f30fe32e7fe837de940dd1d0e).

---

But also if we would add more emitters in the future I thought it would make sense to enforce the use of the self emitter, should we really ignore it?

---

Ah yes, and velocity doesn't track the location so we can't use that packet. 

It's API contract for unsupported adventure methods to not throw exceptions.

---

Alright, then I'll make it silent was well

---

Are null checks exempt, cause I see them in some places or should everything fail silently?

---

@kezz, what do you say?
(also to the other conversation below)